### ISSUE TITLE
vision-transformer problem report

### ISSUE THREAD

**Issue by ChenDaiwei-99** [pytorch/examples#1184]
![image](https://github.com/pytorch/examples/assets/81737228/940d626c-5101-41ae-a36f-496319c03b0a)
![image](https://github.com/pytorch/examples/assets/81737228/7ea00869-7dd0-48ac-b0ef-1b25d57cbccc)
![image](https://github.com/pytorch/examples/assets/81737228/d3d5b76c-2e5e-4fcb-91c3-aa692938de0c)

**Comment by msaroufim** [comment]
If you wanna contribute some fixes would be happy to merge

**Comment by ChenDaiwei-99** [comment]
Sure, will do it in a few days :)

**Comment by colinosterman** [comment]
I can confirm this example is badly broken.  I added some code to compare individual labels to predictions and discovered the forward pass of ViT always returns the same tensor.  No matter the input. The tensor it returns is different each time I run it, even if I load the same weights from the save file and don't do any training. It's no wonder it can't do better than 2.3. Always giving the same prediction should accidentally hit on the correct label about as often as random guessing.

**Comment by fanqieguo** [comment]
I added the printout of accuracy in the code, but during the training process, the accuracy does not improve, and the loss does not converge, even after training for many epochs. Does this model really work? I think this model has serious problems. I hope to get an answer, this is very important to me
